closure-typing/calculator.js

```diff
@@ -1,5 +1,12 @@
-function calculator() {
-  // your code here
+function calculator(str) {
+  // solution for only one type of operator, but several arguments
+  if (!str) return;
+  let result = (str.includes("+")) ? str.split("+").reduce((res, a)    => res + 1 * a, 0) :
+               (str.includes("-")) ? str.split("-").reduce((res, a, i) => i ? res - a : a, 0) :
+               (str.includes("*")) ? str.split("*").reduce((res, a, i) => i ? res * a : a, 0) :
+               (str.includes("/")) ? str.split("/").reduce((res, a, i) => i && a !== 0 ? res / a : a, 0) : 0;
+  return result;
```

Comment:
As I sad really really hard to read and understand code, but its great that you made extra effort and created solution of more than two arguments
hah, and from your code I can say that you would love RegExp's :)